doc/devel/contributing.rst

```diff
@@ -19,6 +19,59 @@ is expected to abide by our
 Below, you can find a number of ways to contribute, and how to connect with the
 Matplotlib community.
 
+.. _start-contributing:
+
+Getting started with contributing?
+==================================
+
+There is no pre-defined pathway for new contributors; you can look at existing
+issue and pull request discussions, and follow the discussions during code
+review. Or you can deep-dive into a subset of code-base to understand what is
+going on.
+
+There are a few typical new contributor profiles:
+
+* **You are a Matplotlib user, and you see a bug, a potential improvement, or
+  something that annoys you, and you can fix it.**
+
+  This is how a lot of people approach open-source development. You can search
+  our issue tracker for an existing issue that describes your problem or open a
+  new issue to inform us of the problem you observed and discuss the best
+  approach to fix it. If your contributions would not be captured on GitHub
```

Comment:
I'm thinking that it is not really required to open an issue before fixing it? I often see people opening an issue the same minute they push a PR. While that can be relevant sometimes, I do not think that is the norm nor what we want to encourage? The wording above *may* lead to that impression, although I' not really sure if it is worthwhile modifying in any way...

Reply:
Fair point. I have an opposite experience in SymPy, but it doesn't seem to be that much of a problem here admittedly.